make error on 10.9

(the formula says `depends_on :macosx => :maverics` so I'm assuming 10.9 should be supported.)

```
~$ brew install hhvm
==> Installing hhvm from hhvm/hhvm
==> Installing dependencies for hhvm/hhvm/hhvm: llvm, dwarfutils, libelf, boost, gflags, glog
==> Installing hhvm/hhvm/hhvm dependency: llvm
==> Downloading http://llvm.org/releases/3.6.2/llvm-3.6.2.src.tar.xz
Already downloaded: /Library/Caches/Homebrew/llvm-3.6.2.tar.xz
==> Downloading http://llvm.org/releases/3.6.2/cfe-3.6.2.src.tar.xz
Already downloaded: /Library/Caches/Homebrew/llvm--clang-3.6.2.tar.xz
==> cmake -G Unix Makefiles /private/tmp/llvm20151218-51218-d3tdvy/llvm-3.6.2.src -DCMAKE_C_FLAGS_RE
==> make
==> make install
==> Caveats
LLVM executables are installed in /usr/local/opt/llvm/bin.
Extra tools are installed in /usr/local/opt/llvm/share/llvm.

This formula is keg-only, which means it was not symlinked into /usr/local.

OS X already provides this software and installing another version in
parallel can cause all kinds of trouble.

Generally there are no consequences of this for you. If you build your
own software and it requires this formula, you'll need to add to your
build variables:

    LDFLAGS:  -L/usr/local/opt/llvm/lib
    CPPFLAGS: -I/usr/local/opt/llvm/include

==> Summary
🍺  /usr/local/Cellar/llvm/3.6.2: 1350 files, 426M, built in 9.4 minutes
==> Installing hhvm/hhvm/hhvm dependency: dwarfutils
==> Downloading https://homebrew.bintray.com/bottles/dwarfutils-20150507.mavericks.bottle.tar.gz
######################################################################## 100.0%
==> Pouring dwarfutils-20150507.mavericks.bottle.tar.gz
🍺  /usr/local/Cellar/dwarfutils/20150507: 8 files, 1008K
==> Installing hhvm/hhvm/hhvm dependency: libelf
==> Downloading https://homebrew.bintray.com/bottles/libelf-0.8.13_1.mavericks.bottle.tar.gz
######################################################################## 100.0%
==> Pouring libelf-0.8.13_1.mavericks.bottle.tar.gz
🍺  /usr/local/Cellar/libelf/0.8.13_1: 11 files, 264K
==> Installing hhvm/hhvm/hhvm dependency: boost
==> Downloading https://homebrew.bintray.com/bottles/boost-1.59.0.mavericks.bottle.1.tar.gz
######################################################################## 100.0%
==> Pouring boost-1.59.0.mavericks.bottle.1.tar.gz
🍺  /usr/local/Cellar/boost/1.59.0: 10928 files, 434M
==> Installing hhvm/hhvm/hhvm dependency: gflags
==> Downloading https://homebrew.bintray.com/bottles/gflags-2.1.2.mavericks.bottle.2.tar.gz
######################################################################## 100.0%
==> Pouring gflags-2.1.2.mavericks.bottle.2.tar.gz
🍺  /usr/local/Cellar/gflags/2.1.2: 16 files, 388K
==> Installing hhvm/hhvm/hhvm dependency: glog
==> Downloading https://github.com/google/glog/archive/v0.3.4.tar.gz
==> Downloading from https://codeload.github.com/google/glog/tar.gz/v0.3.4
######################################################################## 100.0%
==> ./configure --prefix=/usr/local/Cellar/glog/0.3.4
==> make install
🍺  /usr/local/Cellar/glog/0.3.4: 23 files, 500K, built in 16 seconds
==> Installing hhvm/hhvm/hhvm dependency: imagemagick
==> Downloading https://homebrew.bintray.com/bottles/imagemagick-6.9.2-8.mavericks.bottle.tar.gz
######################################################################## 100.0%
==> Pouring imagemagick-6.9.2-8.mavericks.bottle.tar.gz
🍺  /usr/local/Cellar/imagemagick/6.9.2-8: 1453 files, 22M
==> Installing hhvm/hhvm/hhvm dependency: jemalloc
==> Downloading https://homebrew.bintray.com/bottles/jemalloc-4.0.0.mavericks.bottle.tar.gz
######################################################################## 100.0%
==> Pouring jemalloc-4.0.0.mavericks.bottle.tar.gz
🍺  /usr/local/Cellar/jemalloc/4.0.0: 14 files, 1.4M
==> Installing hhvm/hhvm/hhvm dependency: libevent
==> Downloading https://homebrew.bintray.com/bottles/libevent-2.0.22.mavericks.bottle.tar.gz
######################################################################## 100.0%
==> Pouring libevent-2.0.22.mavericks.bottle.tar.gz
🍺  /usr/local/Cellar/libevent/2.0.22: 48 files, 1.8M
==> Installing hhvm/hhvm/hhvm dependency: memcached
==> Downloading https://homebrew.bintray.com/bottles/memcached-1.4.24.mavericks.bottle.tar.gz
######################################################################## 100.0%
==> Pouring memcached-1.4.24.mavericks.bottle.tar.gz
==> Caveats
To have launchd start memcached at login:
  ln -sfv /usr/local/opt/memcached/*.plist ~/Library/LaunchAgents
Then to load memcached now:
  launchctl load ~/Library/LaunchAgents/homebrew.mxcl.memcached.plist
Or, if you don't want/need launchctl, you can just run:
  /usr/local/opt/memcached/bin/memcached
==> Summary
🍺  /usr/local/Cellar/memcached/1.4.24: 10 files, 192K
==> Installing hhvm/hhvm/hhvm dependency: libmemcached
==> Downloading https://homebrew.bintray.com/bottles/libmemcached-1.0.18_1.mavericks.bottle.tar.gz
######################################################################## 100.0%
==> Pouring libmemcached-1.0.18_1.mavericks.bottle.tar.gz
🍺  /usr/local/Cellar/libmemcached/1.0.18_1: 227 files, 2.2M
==> Installing hhvm/hhvm/hhvm dependency: libzip
==> Downloading https://homebrew.bintray.com/bottles/libzip-1.0.1.mavericks.bottle.tar.gz
######################################################################## 100.0%
==> Pouring libzip-1.0.1.mavericks.bottle.tar.gz
🍺  /usr/local/Cellar/libzip/1.0.1: 118 files, 696K
==> Installing hhvm/hhvm/hhvm dependency: lz4
==> Downloading https://homebrew.bintray.com/bottles/lz4-r131.mavericks.bottle.tar.gz
######################################################################## 100.0%
==> Pouring lz4-r131.mavericks.bottle.tar.gz
🍺  /usr/local/Cellar/lz4/r131: 12 files, 440K
==> Installing hhvm/hhvm/hhvm dependency: mhash
==> Downloading https://homebrew.bintray.com/bottles/mhash-0.9.9.9.mavericks.bottle.tar.gz
######################################################################## 100.0%
==> Pouring mhash-0.9.9.9.mavericks.bottle.tar.gz
🍺  /usr/local/Cellar/mhash/0.9.9.9: 17 files, 508K
==> Installing hhvm/hhvm/hhvm dependency: mcrypt
==> Downloading https://homebrew.bintray.com/bottles/mcrypt-2.6.8.mavericks.bottle.1.tar.gz
######################################################################## 100.0%
==> Pouring mcrypt-2.6.8.mavericks.bottle.1.tar.gz
🍺  /usr/local/Cellar/mcrypt/2.6.8: 15 files, 452K
==> Installing hhvm/hhvm/hhvm dependency: oniguruma
==> Downloading https://homebrew.bintray.com/bottles/oniguruma-5.9.6.mavericks.bottle.1.tar.gz
######################################################################## 100.0%
==> Pouring oniguruma-5.9.6.mavericks.bottle.1.tar.gz
🍺  /usr/local/Cellar/oniguruma/5.9.6: 12 files, 1.1M
==> Installing hhvm/hhvm/hhvm
==> Downloading http://dl.hhvm.com/source/hhvm-3.11.0.tar.bz2
######################################################################## 100.0%
==> cmake -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/hhvm/3.11.0 -DDEFAULT_CONFIG_DIR=/usr/local/etc/hhvm -D
==> make
Last 15 lines from /Users/erik/Library/Logs/Homebrew/hhvm/02.make:
make[5]: *** [mysys_ssl/CMakeFiles/mysys_ssl.dir/my_aes.cc.o] Error 1
make[5]: *** [mysys_ssl/CMakeFiles/mysys_ssl.dir/my_md5.cc.o] Error 1
1 error generated.
make[5]: *** [mysys_ssl/CMakeFiles/mysys_ssl.dir/crypt_genhash_impl.cc.o] Error 1
1 error generated.
make[5]: *** [mysys_ssl/CMakeFiles/mysys_ssl.dir/my_rnd.cc.o] Error 1
1 error generated.
make[5]: *** [mysys_ssl/CMakeFiles/mysys_ssl.dir/my_default.cc.o] Error 1
1 error generated.
make[5]: *** [mysys_ssl/CMakeFiles/mysys_ssl.dir/my_getopt.cc.o] Error 1
make[4]: *** [mysys_ssl/CMakeFiles/mysys_ssl.dir/all] Error 2
make[3]: *** [all] Error 2
make[2]: *** [third-party/webscalesqlclient/build/src/webscalesqlclient-stamp/webscalesqlclient-build] Error 2
make[1]: *** [third-party/webscalesqlclient/CMakeFiles/webscalesqlclient.dir/all] Error 2
make: *** [all] Error 2

READ THIS: https://git.io/brew-troubleshooting
If reporting this issue please do so at (not Homebrew/homebrew):
  https://github.com/hhvm/hhvm/issues 
```

```
[  0%] No download step for 'webscalesqlclient'
In file included from /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/re2/src/util/hash.cc:41:
/tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/re2/src/util/util.h:23:10: fatal error: 'ctime' file not found
#include <ctime>
         ^

[  0%] Building C object third-party/libafdt/CMakeFiles/afdt.dir/src/sync.c.o
/tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/double-conversion/src/double-conversion/cached-powers.cc:134:18: warning: unused variable 'kCachedPowersLength' [-Wunused-const-variable]
static const int kCachedPowersLength = ARRAY_SIZE(kCachedPowers);
                 ^

[  2%] Building C object third-party/timelib/CMakeFiles/timelib.dir/unixtime2tm.c.o
In file included from hphp.y:11:
/tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/hphp/compiler/parser/parser.h:20:10: fatal error: 'functional' file not found
#include <functional>
         ^

[  2%] Building CXX object hphp/parser/CMakeFiles/hphp_parser.dir/scanner.cpp.o
-- Check for working CXX compiler: /usr/local/opt/llvm/bin/clang++
In file included from /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/hphp/parser/parser.cpp:16:
/tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/hphp/parser/parser.h:20:10: fatal error: 'map' file not found
#include <map>
         ^
In file included from /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/hphp/parser/scanner.cpp:16:
/tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/hphp/parser/scanner.h:20:10: fatal error: 'sstream' file not found
#include <sstream>
         ^

51 libdyld.dylib            0x00007fff8831a5fd start + 1
Stack dump:
0.	Program arguments: /usr/local/Cellar/llvm/3.6.2/bin/clang-3.6 -cc1 -triple x86_64-apple-macosx10.9.0 -emit-obj -disable-free -main-file-name hphp.5.tab.cpp -mrelocation-model pic -pic-level 2 -mthread-model posix -mdisable-fp-elim -masm-verbose -munwind-tables -target-cpu core2 -target-feature +sse4.2 -target-linker-version 241.9 -dwarf-column-info -coverage-file /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/hphp/parser/CMakeFiles/hphp_parser.dir/hphp.5.tab.cpp.o -resource-dir /usr/local/Cellar/llvm/3.6.2/bin/../lib/clang/3.6.2 -D DEFAULT_CONFIG_DIR="/usr/local/etc/hhvm" -D ENABLE_FASTCGI=1 -D FOLLY_HAVE_PTHREAD_ATFORK=1 -D FOLLY_HAVE_SCHED_H=1 -D FOLLY_HAVE_VLA=1 -D FOLLY_HAVE_WEAK_SYMBOLS=1 -D FOLLY_NO_CONFIG=1 -D FOLLY_USE_LIBCPP=1 -D HAVE_BOOST1_49 -D HAVE_CURL_MULTI_WAIT -D HAVE_LIBDL -D HHVM -D HHVM_DYNAMIC_EXTENSION_DIR="/usr/local/Cellar/hhvm/3.11.0/lib/hhvm/extensions" -D HPHP_OSS=1 -D MACOSX_DEPLOYMENT_TARGET=10.6 -D MAC_OS_X_VERSION_MIN_REQUIRED=MAC_OS_X_VERSION_10_6 -D MBFL_STATIC -D NDEBUG -D NO_HARDWARE_COUNTERS=1 -D NO_LIB_GFLAGS -D NO_TCMALLOC=1 -D PACKAGE=hhvm -D PACKAGE_VERSION=Release -D PHP_MYSQL_UNIX_SOCK_ADDR="/tmp/mysql.sock" -D RELEASE=1 -D SKIP_USER_CHANGE=1 -D THRIFT_MUTEX_EMULATE_PTHREAD_TIMEDLOCK -D USE_CMAKE -D USE_EDITLINE -D USE_JEMALLOC=1 -D _PTHREADS=1 -D _REENTRANT=1 -D __STDC_FORMAT_MACROS -D NDEBUG -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0 -I /usr/local/include -I /usr/local/opt/glog/include -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/re2/src -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/squangle/src -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/webscalesqlclient/src/include -I /usr/local/opt/pcre/include -I /usr/local/opt/libevent/include -I /usr/include/libxml2 -I /usr/local/opt/lz4/include -I /usr/local/opt/libzip/include -I /usr/local/opt/libzip/lib/libzip/include -I /usr/local/opt/icu4c/include -I /usr/local/opt/jemalloc/include -I /usr/local/opt/openssl/include -I /usr/local/opt/libelf/include/libelf -I /usr/include/security -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/hphp -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/double-conversion/src -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/fastlz -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/timelib -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/libafdt/src -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/libmbfl -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/libmbfl/mbfl -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/libmbfl/filters -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/proxygen/src -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/folly -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/folly/src -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/thrift/src -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party/wangle/src -I /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/third-party -I /usr/local/opt/readline/include -stdlib=libc++ -O3 -Wall -Woverloaded-virtual -Wno-deprecated -Wno-strict-aliasing -Wno-write-strings -Wno-invalid-offsetof -Wno-error=array-bounds -Wno-error=switch -Werror=format-security -Wno-unused-result -Wno-sign-compare -Wno-attributes -Wno-maybe-uninitialized -Wno-mismatched-tags -Wno-unknown-warning-option -Wno-return-type-c-linkage -std=gnu++11 -fdebug-compilation-dir /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/hphp/parser -ferror-limit 19 -fmessage-length 0 -fno-operator-names -stack-protector 1 -mstackrealign -fblocks -fobjc-runtime=macosx-10.9.0 -fencode-extended-block-signature -fcxx-exceptions -fexceptions -fmax-type-align=16 -fdiagnostics-show-option -vectorize-loops -vectorize-slp -o CMakeFiles/hphp_parser.dir/hphp.5.tab.cpp.o -x c++ /tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/hphp/parser/hphp.5.tab.cpp 
1.	/tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/hphp/runtime/vm/type-constraint.h:156:7: current parser token ':'
2.	/tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/hphp/runtime/vm/type-constraint.h:34:1: parsing namespace 'HPHP'
3.	/tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/hphp/runtime/vm/type-constraint.h:42:1: parsing struct/union/class body 'TypeConstraint'
4.	/tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/hphp/runtime/vm/type-constraint.h:152:44: parsing function body 'underlyingDataType'
5.	/tmp/hhvm20151223-95148-1492ihp/hhvm-3.11.0/hphp/runtime/vm/type-constraint.h:152:44: in compound statement ('{}')
```

I encountered the same problem on 10.11.

Relevant gist when it errored out: https://gist.github.com/tengyifei/6c76193c105b71f9c8d4

```
/tmp/hhvm20160106-21041-mxfygo/hhvm-3.11.0/third-party/webscalesqlclient/mysql-5.6/include/my_global.h:789:12: fatal error: 'cmath' file not found
  #include <cmath>
```

--- a/hhvm.rb
+++ b/hhvm.rb
@@ -18,7 +18,7 @@ class Hhvm < Formula
   # We need to build with upstream clang -- the version Apple ships doesn't
   # support TLS, which HHVM uses heavily. (And gcc compiles HHVM fine, but
   # causes ld to trip an assert and fail, for unclear reasons.)
-  depends_on "llvm" => [:build, "with-clang"]
+  depends_on "llvm" => [:build, "with-clang", "with-libcxx"]
 
   depends_on "autoconf" => :build
   depends_on "automake" => :build

